tools/codegen/dest/register_dispatch_key.py

```diff
@@ -169,9 +169,12 @@ def generate_defn(cpp_sig: CppSignature) -> str:
                     )
 
                     # Only tensor like arguments are eligible
-                    device_of = next((f'{a.name}' for a in candidate_args if a.type.is_tensor_like()), None)
-                    if device_of is not None:
-                        device_guard = f"const OptionalDeviceGuard device_guard(device_of({device_of}));"
+                    device_guard = 'c10::optional<Device> common_device = nullopt;'
+                    for arg in candidate_args:
```

Comment:
This ofen made `self` generated twice. Needs to be fixed.

Reply:
Since you're checking everything, there's no reason to setup candidate args anymore, since you need to pick up all Tensor-like arguments regardless.